nav2_costmap_2d/plugins/costmap_filters/speed_filter.cpp

```diff
@@ -278,30 +283,48 @@ void SpeedFilter::process(
   } else {
     // Normal case: speed_mask_data in range of [1..100]
     speed_limit_ = speed_mask_data * multiplier_ + base_;
-    if (speed_limit_ < 0.0) {
-      RCLCPP_WARN(
-        logger_,
-        "SpeedFilter: Speed limit in filter_mask[%i, %i] is less than 0%%, "
-        "which can not be true. Setting it to 0%% value.",
-        mask_robot_i, mask_robot_j);
-      speed_limit_ = NO_SPEED_LIMIT;
-    }
-    if (speed_limit_ > 100.0) {
-      RCLCPP_WARN(
-        logger_,
-        "SpeedFilter: Speed limit in filter_mask[%i, %i] is higher than 100%%, "
-        "which can not be true. Setting it to 100%% value.",
-        mask_robot_i, mask_robot_j);
-      speed_limit_ = 100.0;
+    if (percentage_) {
+      if (speed_limit_ < 0.0) {
```

Comment:
The < 0 and > 100 can be combined into 1 check since they both have the same outcome:

`if (speed_limit > 100.0 || speed_limit < 0.0) {log: "SpeedFilter: Speed limit in filter_mask[...] is %f, out of bounds of [0, 100]. Setting it to no-limit value."}`

Reply:
Done starting from https://github.com/ros-planning/navigation2/pull/2149/commits/6bd018bee773e8ae042062cde5963646a93cba25#diff-d344e376ecbc8e5ecd5045731e13855c6d2a9a955cdaee9b31dddbbee7681466L287